arch/risc-v/src/common/riscv_internal.h

```diff
@@ -356,40 +356,24 @@ void riscv_cpu_boot(int cpu);
 int riscv_smp_call_handler(int irq, void *c, void *arg);
 #endif
 
+#ifdef CONFIG_ARCH_RV_CPUID_MAP
 /****************************************************************************
- * Name: riscv_mhartid
+ * Name: riscv_mhartid / riscv_hartid_to_cpuid / riscv_cpuid_to_hartid
  *
  * Description:
- *   Context aware way to query hart id (physical core ID)
- *
- * Returned Value:
- *   Hart id
+ *   CPU ID mapping functions for systems where physical hart IDs don't match
+ *   logical CPU IDs.
  *
  ****************************************************************************/
 
 uintptr_t riscv_mhartid(void);
```

Comment:
This should always return the physical core ID, so it does not depend on CONFIG_ARCH_RV_CPUID_MAP